pf4j/src/test/java/org/pf4j/DefaultPluginManagerTest.java

```diff
@@ -190,4 +190,44 @@ public void deleteJarPlugin() throws Exception {
         assertFalse(pluginJar.file().exists());
     }
 
+    @Test
+    public void loadingPluginWithMissingDependencyDoesNotBreakOtherPlugins() throws IOException {
+        // Load 2 plugins, one with a dependency that is missing and one without any dependencies.
+        PluginZip pluginZip1 = new PluginZip.Builder(pluginsPath.resolve("my-first-plugin-1.1.1.zip"), "myPlugin1")
+            .pluginVersion("1.1.1")
+            .pluginDependencies("myBasePlugin")
+            .build();
+
+        PluginZip pluginZip2 = new PluginZip.Builder(pluginsPath.resolve("my-second-plugin-2.2.2.zip"), "myPlugin2")
+            .pluginVersion("2.2.2")
+            .build();
+
+        pluginManager.loadPlugins();
+        pluginManager.startPlugins();
+
+        // myPlugin2 should have been started at this point.
+        assertEquals(PluginState.STARTED, pluginManager.getPlugin(pluginZip2.pluginId()).getPluginState());
```

Comment:
Right now, the test will fail at this point.

Reply:
Some suggested fixes pushed in https://github.com/pf4j/pf4j/pull/558/commits/b96bb93b8ceb44a43c6bb95df3057bbd59c471f6.